src/actions/expressions.js

```diff
@@ -98,6 +100,7 @@ function evaluateExpression(expression, frameId: frameIdType) {
     return dispatch({
       type: constants.EVALUATE_EXPRESSION,
       input: expression.input,
+      visible: expression.visible == undefined ? true : expression.visible,
```

Comment:
hmm, i wonder if we should force `expression.visible` to be a boolean and then not worry about these other states... Flow to the rescue?